File: rest-service/src/main/java/mb/lib/formatter/db/DeleteUserReportLink.java
package mb.lib.formatter.db;

import java.sql.Connection;
import java.sql.PreparedStatement;

import io.vulpine.lib.query.util.basic.BasicPreparedVoidQuery;
import mb.lib.model.HashID;

public class DeleteUserReportLink
{
  private static final String Query = """
    DELETE FROM
      userlogins5.multiblast_users_to_fmt_jobs
    WHERE
      report_digest = ?
      AND user_id = ?
    """;

  private final Connection con;
  private final HashID reportID;
  private final long userID;

  public DeleteUserReportLink(Connection con, HashID reportID, long userID) {
    this.con      = con;
    this.reportID = reportID;
    this.userID   = userID;
  }

  public void run() throws Exception {
    new BasicPreparedVoidQuery(Query, con, this::prepare).execute();
  }

  private void prepare(PreparedStatement ps) throws Exception {
    ps.setBytes(1, reportID.bytes());
    ps.setLong(2, userID);
  }
}

File: rest-service/src/main/kotlin/mb/lib/report/db/DeleteUserReportLink.kt
package mb.lib.report.db

import io.vulpine.lib.query.util.basic.BasicPreparedVoidQuery
import mb.lib.model.HashID
import java.sql.Connection
import java.sql.PreparedStatement

data class DeleteUserReportLink(
  val con: Connection,
  val reportID: HashID,
  val userID: Long
) {
  companion object {
    private const val Query = """
    DELETE FROM
      userlogins5.multiblast_users_to_fmt_jobs
    WHERE
      report_digest = ?
      AND user_id = ?
    """
  }
  fun run() = BasicPreparedVoidQuery(Query, con, this::prepare).execute()

  private fun prepare(ps: PreparedStatement) {
    ps.setBytes(1, reportID.bytes)
    ps.setLong(2, userID)
  }
}
